shadows/framework/src/main/java/org/robolectric/shadows/ShadowInputMethodManager.java

```diff
@@ -274,6 +275,12 @@ public static void reset() {
     if (apiLevel > P) {
       _reflector.getInstanceMap().clear();
     }
+    softInputVisible = false;
+    visibilityChangeHandler = Optional.absent();
+    privateCommandListener = Optional.absent();
+    inputMethodInfoList = ImmutableList.of();
```

Comment:
> We need to clear two list instanes.

Since the lists are immutable, maybe we need to reset them to new instances rather than attempting to clear them directly. i have tried clear also but tests were failing